lib/flowmailer_rails/mailer.rb

```diff
@@ -39,7 +40,13 @@ def submit_message(json)
         response = api_client.post(path, json, authorization_header)
 
         raise ExpiredAccessTokenError if response.status == 401
-        raise DeliveryError, response.body unless response.success?
+
+        unless response.success?
+          too_big_error = response.body&.dig("allErrors")&.find { |error| error["code"] == "message.toobig" }
+          raise TooBigMessageError, too_big_error["defaultMessage"] if too_big_error.present?
+
+          raise DeliveryError, response.body
+        end
```

Comment:
Could we maybe extract this to a separate method like `handleError/failure/something` to make the `submit_message` method cleaner?